python/pyspark/tests/test_context.py

```diff
@@ -273,7 +273,8 @@ def setUp(self):
         self.tempFile.close()
         os.chmod(self.tempFile.name, stat.S_IRWXU | stat.S_IXGRP | stat.S_IRGRP |
                  stat.S_IROTH | stat.S_IXOTH)
-        conf = SparkConf().set("spark.driver.resource.gpu.amount", "1")
+        conf = SparkConf().set("spark.test.home", SPARK_HOME)
```

Comment:
This only passed on my laptop locally, but always failed on Jenkins:

```
Py4JJavaError: An error occurred while calling None.org.apache.spark.api.java.JavaSparkContext.
: java.lang.AssertionError: assertion failed: spark.test.home is not set!
	at scala.Predef$.assert(Predef.scala:223)
	at org.apache.spark.resource.ResourceUtils$.$anonfun$getOrCreateResourcesDir$1(ResourceUtils.scala:321)
	at scala.Option.getOrElse(Option.scala:138)
	at org.apache.spark.resource.ResourceUtils$.getOrCreateResourcesDir(ResourceUtils.scala:319)
	at org.apache.spark.resource.ResourceUtils$.acquireLock(ResourceUtils.scala:279)
	at org.apache.spark.resource.ResourceUtils$.acquireResources(ResourceUtils.scala:128)
	at org.apache.spark.SparkContext.<init>(SparkContext.scala:399)
	at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
	at sun.reflect.GeneratedConstructorAccessor17.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:238)
	at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
	at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:748)
```

Any ideas ? @tgravescs @HyukjinKwon

Reply:
This is because `spark.test.home` property is only set in testing mode in JVM. I left a comment https://github.com/apache/spark/pull/25047/files/95111b0b9c0a4732041d19b5e221eecea4408147#r306121558

Reply:
Thank you for your help :)